### PR TITLE
Add 'get-bucket-notification-cfg' subcmd for R2

### PR DESCRIPTION
**Pull request by mattdeboard**
## What this PR solves / how to test

Fixes MQ-599.

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [X] Included
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [X] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [X] Not necessary because: Documentation PR for this new feature is not open yet, but will be landing the evening of Monday, April 1

### CODE REVIEW

File: packages/wrangler/src/r2/helpers.ts [hunk 1]
```diff
@@ -368,6 +376,20 @@ export function eventNotificationHeaders(
...
 }
+
+export async function getEventNotificationConfig(
+	apiCredentials: ApiCredentials,
+	accountId: string,
+	bucketName: string
+): Promise<GetNotificationConfigResponse> {
+	const headers = eventNotificationHeaders(apiCredentials);
+	logger.log(`Fetching notification configuration for bucket ${bucketName}...`);
+	return await fetchResult<GetNotificationConfigResponse>(
```
Comment (petebacondarwin) [inline packages/wrangler/src/r2/helpers.ts:387]: I am not sure that simply returning the value here will result in the response being logged to the console, if that it is what you expected.

File: packages/wrangler/src/r2/helpers.ts [hunk 2]
```diff
@@ -368,6 +376,20 @@ export function eventNotificationHeaders(
...
 	return headers;
 }
+
+export async function getEventNotificationConfig(
+	apiCredentials: ApiCredentials,
+	accountId: string,
+	bucketName: string
+): Promise<GetNotificationConfigResponse> {
+	const headers = eventNotificationHeaders(apiCredentials);
+	logger.log(`Fetching notification configuration for bucket ${bucketName}...`);
```
Comment (petebacondarwin) [inline packages/wrangler/src/r2/helpers.ts:386]: In other commands you are sending metrics events, did you want to do that here too?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]

File: packages/wrangler/src/r2/helpers.ts (hunk 2 above)
Comment (mattdeboard) [reply packages/wrangler/src/r2/helpers.ts:447]: Can you link to an example of what you mean?

File: packages/wrangler/src/r2/helpers.ts (hunk 1 above)
Comment (mattdeboard) [reply packages/wrangler/src/r2/helpers.ts:448]: Yeah, I have expanded on this so the response is formatted into a table and printed via `logger.table`

File: .changeset/rare-toys-wave.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+"wrangler": minor
+---
+
+Add command for fetching R2 Event Notification configurations for a given bucket
```
Comment (RamIdeas) [inline .changeset/rare-toys-wave.md:5]: Can you make this changeset conform to the format in CONTRIBUTING.md please

Comment (mattdeboard) [reply .changeset/rare-toys-wave.md:5]: What specifically needs added? Just `feature:` at the beginning?

Comment (petebacondarwin) [reply .changeset/rare-toys-wave.md:5]: Yep....

```suggestion
feature: add command for fetching R2 Event Notification configurations for a given bucket
```

File: packages/wrangler/src/r2/helpers.ts (hunk 2 above)
Comment (petebacondarwin) [reply packages/wrangler/src/r2/helpers.ts:447]: https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/r2/index.ts#L448-L450

Comment (petebacondarwin) [reply packages/wrangler/src/r2/helpers.ts:447]: Not a blocker for this PR, and it is up to you whether to you want to add metrics for these commands.

**Review by petebacondarwin — APPROVED** [review]
LGTM!

File: .changeset/rare-toys-wave.md (hunk 3 above)
Comment (mattdeboard) [reply .changeset/rare-toys-wave.md]: Got it, done

File: packages/wrangler/src/r2/helpers.ts (hunk 2 above)
Comment (mattdeboard) [reply packages/wrangler/src/r2/helpers.ts:447]: Oh, wow, nice, I did not know that was in there. That is actually a great idea but I'm going to make a separate work ticket to do it for all the notification handlers

**Pull request merged**